### ISSUE TITLE
🐛 `nearest_tau` still returns the wrong basin when the scan under-resolves the curve

### ISSUE THREAD

**Issue by nstarman** [GalacticDynamics/coordinax#847]
Follow-up to #841, which fixed a *different* failure of the same function. #841 stopped bisection converging onto a local **maximum**. The remaining defect returns a genuine local **minimum** — just not the right one — and neither guard #841 added can see it.

## Reproducer

```python
def wiggly(tau):                      # 32 wiggles across the bounds
    t = tau.ustrip("s")
    return u.Q(jnp.stack([t, 0.3 * jnp.sin(20 * t), jnp.zeros_like(t)]), "km")

b = cxfc.FrenetSerretBuilder(wiggly, "s")
x = u.Q(jnp.asarray([8.45268, -0.10722, 0.0]), "km")
cxfc.nearest_tau(b, x, bounds=(u.Q(0.0, "s"), u.Q(10.0, "s")))   # default n_seed=64
```

| | |
|---|---|
| returned `tau` | 8.34701157, distance 0.107439 km |
| true nearest `tau` | 8.463675, distance 0.011169 km |
| | **9.6x too far** |
| `d²(dist²)/dτ²` at the returned point | **+59.200** — a genuine local *minimum* |

3 of 12 random near-curve queries on this curve are silently wrong.

## Why #841s guards cannot catch it

#841 added two checks, and the returned point satisfies both:

1. **Orientation** (`bracket_has_minimum`): the point is a minimum, not a maximum, so the `(r_lo > 0) & (r_hi < 0)` test passes.
2. **No worse than the scan seed**: `dist2(returned) = 0.011544` against `d_seed = 0.036942`, so the post-check passes too.

The true minimum lies *between* two seeds, both of which have a larger distance than the seed the scan picked. The coarse argmin is therefore in the wrong basin, and no check applied to the returned answer can detect that — the answer really is the best point in the basin it was given.

This is the docstring assumption made explicit in #841 ("the argmin is within one spacing of the true minimiser **provided `n_seed` resolves the curve**") failing in a way that does not raise. At `n_seed=128` the same curve resolves and other queries answer correctly, so raising `n_seed` remains a valid remedy — but the failure is silent, which is the problem.

## Not a duplicate of #841

#841s own claim — that a bracket satisfying the endpoint orientation cannot converge onto the maximum next door — is true and verified. This issue is the separate basin-selection failure underneath it, which the audit finding originally attributed to the maximum path.

Found during an independent review of the follow-up design spec; reproduced twice.

**Comment by nstarman** [comment]
**Correction to my own diagnosis above.** I wrote that the coarse argmin lands in the wrong basin. It does not, and measuring it says something more specific.

The argmin seed is at `tau = 8.41270`, which **is** within one spacing of the true minimum at `8.46368`, and its basin ranks #1 of the 3 local minima the scan finds. The scan is right.

The bracket is the problem:

```
bracket = [8.25397, 8.57143]   width = 0.31746
wiggle period                        = 0.31416
inside the bracket: 2 minima at [8.34701, 8.46367]
                    2 maxima at [8.40155, 8.56310]
    tau=8.34701  dist=0.107439   <- returned
    tau=8.46367  dist=0.011169   <- true
```

The bracket is `±spacing` wide, so it is 2 x spacing across — and once the curve varies on the scale of one spacing, that swallows a whole period. Bisection then converges to *a* root satisfying the endpoint orientation, which is the worse of the two minima inside. This is precisely the caveat added to #841 in review: the orientation test says the residual falls across the bracket, not that the bracket holds a single stationary point.

So "wrong basin" is wrong. It is the **wrong stationary point inside the correct bracket**, which also means searching additional local-minimum seeds would not have helped — the right seed was already chosen.

**The fix that does work** is to refine before bracketing: a second scan inside `[tau0 - spacing, tau0 + spacing]` narrows the bracket by a factor of `n_seed`, and the refined bracket then contains exactly one minimum:

```
refined argmin tau1 = 8.46561, fine bracket = [8.46057, 8.47065], width 0.01008
  minima inside: 1, at 8.46367   <- correct
```

Cost is one extra `vmap` of the distance over `n_seed` points, which only calls `builder.location` rather than the transport solve.

Title and labels unchanged — the defect and the reproducer stand exactly as filed; only the mechanism was misattributed.
